app/modules/tunnel/tunnel_router.py

```
@@ -0,0 +1,73 @@
+    if len(workspace_id) != 16 or not all(c in "0123456789abcdef" for c in workspace_id.lower()):
+        raise HTTPException(status_code=400, detail="workspace_id must be 16 hex characters")
+
+
+@router.get(
+    "/tunnel/workspace/{workspace_id}",
+    response_model=WorkspaceMetadataResponse,
+    description="Get workspace metadata and socket status.",
+)
+async def get_workspace_tunnel(
+    workspace_id: str,
+    user=Depends(AuthService.check_auth),
```

Ruff (0.15.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)